server/middleware.js

```diff
@@ -0,0 +1,72 @@
+const path = require("path");
+
+/**
+ * This middleware functions allows us to redirect any api request to a file like bundle.js back to the original /bundle.js.
+ * When we don't redirect these files back to their base route,
+ * server side errors are thrown due to the server trying to find the /Puzzle/Selection/bundle.js route, etc. upon page refresh.
+ * @param {Express.Request} req - an ExpressJS variable holding request data from an API call
+ * @param {Express.Response} res - an ExpressJS variable holding response data for an API call.
+ * @param {Function} next - the next function to execute.
+ * @returns - a server redirect if the route is for a redirectFile
+ */
+const redirectBundleManifest = (req, res, next) => {
+  const redirectFileNames = [
+    "bundle.js",
+    "bundle.css",
+    "manifest.json",
+    "favicon.ico",
```

Comment:
Is this still needed because favicon.ico shouldn't exist anymore, I replaced it with puzzle-piece-icon.ico

Reply:
It doesn't look like it. I'll change it here and below